deployments/puppet/README.md

```diff
@@ -0,0 +1,144 @@
+  See the [source attribute](
+  https://puppet.com/docs/puppet/latest/types/file.html#file-attribute-source)
+  of the `file` resource for supported value types.  The default source file is
+  provided by the collector package. (**default:**
+  `file:///etc/otel/collector/agent_config.yaml`)
+
+- `$collector_config_dest`: Destination path of the collector config file on
+  the node.  (**default:** `/etc/otel/collector/agent_config.yaml`)
+
+- `$collector_version`: Version of the collector package to install.
+  (**default:** `latest`)
+
+- `splunk_realm`: Which realm to send the data to.  The `SPLUNK_REALM`
+  environment variable will be set with this value for the collector service.
+  (**default:** `us0`)
+
+- `splunk_ingest_url`: The Splunk ingest URL, e.g.
+  `https://ingest.us0.signalfx.com`.  The `SPLUNK_INGEST_URL` environment
+  variable will be set with this value for the collector service. (**default:**
+  `https://ingest.${splunk_realm}.signalfx.com`)
+
+- `$splunk_api_url`: The Splunk API URL, e.g. `https://api.us0.signalfx.com`.
+  The `SPLUNK_API_URL` environment variable will be set with this value for the
+  collector service.  (**default:** `https://api.${splunk_realm}.signalfx.com`)
+
+- `$splunk_trace_url`: The Splunk trace endpoint URL, e.g.
+  `https://ingest.us0.signalfx.com/v2/trace`.  The `SPLUNK_TRACE_URL`
+  environment variable will be set with this value for the collector service.
+  (**default:** `${splunk_ingest_url}/v2/trace`)
+
+- `$splunk_hec_url`: The Splunk HEC endpoint URL, e.g.
+  `https://ingest.us0.signalfx.com/v1/log`.  The `SPLUNK_HEC_URL` environment
+  variable will be set with this value for the collector service.
+  (**default:** `${splunk_ingest_url}/v1/log`)
+
+- `$splunk_hec_token`: The Splunk HEC authentication token.  The
+  `SPLUNK_HEC_TOKEN` environment variable will be set with this value for the
+  collector service.  (**default:** `$splunk_access_token`)
+
+- `$splunk_bundle_dir`: The path to the Smart Agent bundle directory.  The
+  default path is provided by the collector package.  If the specified path is
+  changed from the default value, the path should be an existing directory on
+  the node.  The `SPLUNK_BUNDLE_DIR` environment variable will be set to this
+  value for the collector service.  (**default:**
+  `/usr/lib/splunk-otel-collector/agent-bundle`)
+
+- `$splunk_collectd_dir`: The path to the collectd config directory for the
+  Smart Agent bundle.  The default path is provided by the collector package.
+  If the specified path is changed from the default value, the path should be
+  an existing directory on the node.  The `SPLUNK_COLLECTD_DIR` environment
+  variable will be set to this value for the collector service.  (**default:**
+  `${splunk_bundle_dir}/run/collectd`)
+
+- `$package_stage`: The collector package repository stage to use: `release`,
+  `beta`, or `test`.  (**default:** `release`)
```

Comment:
This is a carry-over from the smart agent since we would occasionally publish beta/test packages.  Maybe we'll eventually do the same for the collector?  I'll just remove it from the doc for now until we actually do.